django/cantusdb_project/main_app/signals.py

```diff
@@ -201,3 +205,64 @@ def update_prefix_field(instance):
         instance.__class__.objects.filter(pk=pk).update(prefix=prefix)
     else:  # feast_code is None, ""
         instance.__class__.objects.filter(pk=pk).update(prefix="")
+
+
+def update_chant_incipit_field(chant: Chant):
+    """Update the incipit field of the specified Chant to be the first
+    several words of the chant's standardized-spelling fulltext
+
+    Args:
+        chant (Chant): The chant from the database whose `incipit` field
+        is to be updated
+    """
+    fulltext: Optional[str] = chant.manuscript_full_text_std_spelling
+    if fulltext:  # many chants in the database have only an incipit -
+        # we should only update the incipit if the chant has a fulltext,
+        # just in case a chant manages to get saved without a fulltext somehow
+        new_incipit: str = generate_incipit(fulltext)
+        Chant.objects.filter(id=chant.id).update(incipit=new_incipit)
+
+
+def update_sequence_incipit_field(sequence: Sequence):
+    """Update the incipit field of the specified Sequence to be the first
+    several words of the sequence's standardized-spelling fulltext
+
+    Args:
+        sequence (Sequence): The sequence from the database whose `incipit`
+        field is to be updated
+    """
+    fulltext: Optional[str] = ""
+    std_fulltext: Optional[str] = sequence.manuscript_full_text_std_spelling
+    ms_fulltext: Optional[str] = sequence.manuscript_full_text
+    title: Optional[str] = sequence.title
+    if title:  # most (all?) sequences in the database have no
```

Comment:
Just checked...

All sequences have empty `manuscript_full_text` and `manuscript_full_text_std_spelling` fields

Reply:
I don't know what Debra et al. have in mind for touching up the Bower database, but I don't find it hard to imagine a situation where someone adds fulltexts for some of our sequences. Or was this a comment on the comment itself?

I'm never quite sure how to handle observations about the state of the live database at particular moments in time. Would something like `# as of late February 2024, all sequences in the database have no fulltext, etc.` be better here, you think? At the very least, it will remain accurate into the future...

Reply:
So, first there is the thing about the comment...we do know that no sequences have full text so I think we can say that here. 

Second, I don't know that it makes sense to use these other text fields for the incipit when we don't currently make use of those text fields. Yes, someday those may be used, but since we currently don't, it seems like it is more likely to cause issues (someone accidentally types something into one of these fields and it messes up the incipit) than not.

Reply:
Would it be better to get rid of the various `if`s here and just set incipit to be equal to `title`?

Reply:
I just pushed commits updating the comment, and always setting the sequence's incipit to its title

Reply:
I think that would be good. Or, as I read it again now, if you wanted to write this forward-looking functionality, maybe just change the `fulltext` variable name to something like `text_to_gen_incipit` or some such. I think setting `fulltext` multiple times to things that are not always "full texts" adds to the confusion.

Reply:
or `text_for_generating_incipit`, since characters are cheap (though I'm sure @JoyfulGen would appreciate being immortalized in the CantusDB codebase in this way :P)

Moot point now though - since the things in the `title` field are already incipits, I've simply gone with `incipit`.